src/quick-lint-js/fe/diagnostic-types.h

```diff
@@ -2288,6 +2288,67 @@
           QLJS_TRANSLATABLE("upper case letters compared with toLowerCase"),   \
           span_operator))                                                      \
                                                                                \
+  QLJS_DIAG_TYPE(                                                              \
+      diag_pointless_strict_comp_against_array_literal, "E0301",               \
+      diagnostic_severity::warning, { source_code_span equals_operator; },     \
+      MESSAGE(                                                                 \
+          QLJS_TRANSLATABLE(                                                   \
+              "using '{0}' against an array literal does not compare items"),  \
+          equals_operator))                                                    \
+                                                                               \
+  QLJS_DIAG_TYPE(                                                              \
+      diag_pointless_comp_against_arrow_function, "E0302",                     \
+      diagnostic_severity::warning,                                            \
+      {                                                                        \
+        source_code_span equals_operator;                                      \
+        string8_view condition;                                                \
+      },                                                                       \
+      MESSAGE(QLJS_TRANSLATABLE("using '{0}' against an arrow "                \
+                                "function always returns '{1}'"),              \
+              equals_operator, condition))                                     \
```

Comment:
Nit: I think `result` or `comparison_result` is a better name than `condition`:
```suggestion
      {                                                                        \
        source_code_span equals_operator;                                      \
        string8_view result;                                                   \
      },                                                                       \
      MESSAGE(QLJS_TRANSLATABLE("using '{0}' against an arrow "                \
                                "function always returns '{1}'"),              \
              equals_operator, result))                                        \
```